Update landing pages for backup charm documents (#55)
canonical/backup-operators: docs/reference/index.md

# Reference

> Technical specifications and architectural details for Baculum charms

Reviewer: For my understanding, is it correct to use both "Bacula" and "Baculum" to describe the charms?

Author: Baculum is part of the Bacula charm system. I think using "backup charms" here would probably be more appropriate. Updated, thanks for noticing it.

serve as authoritative look-up material when configuring,  extending,
or integrating the charm.

## Charm configuration and operations

Operators control charm behavior through configuration options and Juju
actions. Here are the configuration and day-to-day operation related to

> backup charms.

Reviewer: ```suggestion
the Bacula charms.
```
nit, to be consistent with the other mentions of the project in this document

> at runtime.

Reviewer: Since you've grouped the architecture docs into their own section (which I think is a good move btw), I think we should remove this sentence about charm architecture from under the "Charm configuration and operations" section. You could move it under the charm architecture section -- I trust your judgement :)

Author: Updated similar to OpenCTI, thanks!

* [Actions](actions.md)
* [Configurations](configurations.md)
* [Integrations](integrations.md)

## Charm architecture and designs

Components and dependencies within the Bacula charms, along with the
architecture decisions made during charm creation.

* [Backup integrator charm architecture](backup-integrator-charm-architecture.md)
* [Bacula server charm architecture](bacula-server-charm-architecture.md)

> * [Bacula file daemon charm architecture](bacula-fd-charm-architecture.md)

Reviewer: These links will be more future-proofed if we use the reference targets instead:

```suggestion
* [Actions](reference_actions)
* [Configurations](reference_configurations)
* [Integrations](reference_integrations)

## Charm architecture and designs

Components and dependencies within the Bacula charms, along with the
architecture decisions made during charm creation.

* [Backup integrator charm architecture](explanation_backup_integrator_charm_architecture)
* [Bacula server charm architecture](explanation_backup_server_charm_architecture)
* [Bacula file daemon charm architecture](explanation_bacula_fd_charm_architecture)
```

```{toctree}

> :maxdepth: 1

Reviewer: Same comment about not displaying the toctree on the page:
```suggestion
:maxdepth: 1
:hidden:
```
